frontend/src/metabase/query_builder/reducers.js

```diff
@@ -558,6 +562,27 @@ export const visibleTimelineIds = handleActions(
   [],
 );
 
+export const visibleTimelineEventIds = handleActions(
+  {
+    [INITIALIZE_QB]: { next: () => [] },
+    [CREATE_TIMELINE_EVENT]: {
+      next: (state, { payload }) => [...state, payload.timelineEvent.id],
+    },
+    [INITIALIZE_VISIBLE_TIMELINE_EVENTS]: {
```

Comment:
@ranquild, here I add an action type just to handle initialization.

Two reasons:

1. to mark that this is about initializing, simply
2. show/hideTimelineEvent are about a single event. So all across the props and such this becomes tighter.

wdyt?